platform-sdk/swirlds-common/src/main/java/com/swirlds/common/threading/CountUpLatch.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.threading;
+
+import static com.swirlds.common.utility.CompareTo.isLessThan;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.Phaser;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Similar to a {@link java.util.concurrent.CountDownLatch}, but counts up instead.
+ */
+public class CountUpLatch {
+
+    private final AtomicLong currentCount;
+    private final Phaser phaser;
```

Comment:
can you add some javadoc to explain how the phaser is used?

Reply:
I no longer use a phaser for this

Reply:
let me know if the current implementation is not clear and I will add more javadocs